portal-web/test/functional/com/liferay/portalweb/macros/asset/assetpublisher/AssetPublisherPortlet.macro

```diff
@@ -513,6 +513,38 @@ definition {
 			}
 		}
 
+		if (isSet(searchTerm)) {
+			Panel.expandPanel(panel = "Asset Entries");
+
+			Click(locator1 = "APConfiguration#ASSET_SELECTION_MANUAL_ASSET_ENTRIES_SELECT_BUTTON");
+
+			var key_assetType = "${assetType}";
+
+			AssertClick(
+				locator1 = "APConfiguration#ASSET_SELECTION_MANUAL_ASSET_ENTRIES_ASSET",
+				value1 = "${assetType}");
+
+			WaitForElementPresent(locator1 = "APConfigurationManual#ASSET_SELECTION_MANUAL_SELECT_ASSET_ENTRIES_DIALOG");
+
+			SelectFrame.selectFrameNoLoading(locator1 = "IFrame#MODAL_BODY");
+
+			Search.searchCP(searchTerm = "${searchTerm}");
+
+			AssertTextEquals(
+				locator1 = "Search#SEARCH_INFO",
+				value1 = "${searchInfo}");
+
+			PortletEntry.selectAll();
+
+			SelectFrameTop();
+
+			SelectFrame(locator1 = "IFrame#CONFIGURATION");
+
+			Button.clickAdd();
+
+			Alert.viewSuccessMessage();
+		}
+
```

Comment:
Should simplify macro by extract common steps

For example,
```
	Panel.expandPanel(panel = "Asset Entries");

	if (isSet(depotName)) {
		Click(
			key_depotName = "${depotName}",
			locator1 = "APConfiguration#ASSET_SELECTION_MANUAL_ASSET_ENTRIES_SELECT_IN_DEPOT");
	}
	else {
		Click(locator1 = "APConfiguration#ASSET_SELECTION_MANUAL_ASSET_ENTRIES_SELECT_BUTTON");
	}

	var key_assetType = "${assetType}";

	AssertClick(
		locator1 = "APConfiguration#ASSET_SELECTION_MANUAL_ASSET_ENTRIES_ASSET",
		value1 = "${assetType}");

	WaitForElementPresent(locator1 = "APConfigurationManual#ASSET_SELECTION_MANUAL_SELECT_ASSET_ENTRIES_DIALOG");

	SelectFrame.selectFrameNoLoading(locator1 = "IFrame#MODAL_BODY");

	if (isSet(assetTitle)) {
			var key_assetTitle = "${assetTitle}";

			if (isSet(viewNoRelatedAsset)) {
				AssertElementNotPresent(
					key_rowEntry = "${assetTitle}",
					locator1 = "ContentRow#ENTRY_CONTENT_ENTRY_CHECKBOX");
			}
			else {
				WaitForElementPresent(
					key_rowEntry = "${assetTitle}",
					locator1 = "ContentRow#ENTRY_CONTENT_ENTRY_CHECKBOX");

				Click(
					key_rowEntry = "${assetTitle}",
					locator1 = "ContentRow#ENTRY_CONTENT_ENTRY_CHECKBOX");
                                }
        }

        if (isSet(searchTerm)) {
                       Search.searchCP(searchTerm = "${searchTerm}");
                       
                       AssertTextEquals(
                       locator1 = "Search#SEARCH_INFO",
                       value1 = "${searchInfo}");
                       
                       PortletEntry.selectAll();
        }
        
        SelectFrameTop();

	SelectFrame(locator1 = "IFrame#CONFIGURATION");

	Button.clickAdd();

	Alert.viewSuccessMessage();
```

Reply:
Thanks Tim, I was having some trouble to figure out it, I will fix this macro and resend to you.

Reply:
Hi @Tim-Cao! In other to do this, shouldn't we encapsulate the code you mention in another if in other to not be called when searchTerm or assetTitle are not set? I saw some examples that use the method without those parameters. In that cases maybe the code will end up with lots of nested ifs and it's gonna be difficult to read. I see your concern but maybe it is better to duplicate some lines of code but make the code easier to read. What do you think?

Reply:
You’re right. Perfect!